interface/forms/newpatient/common.php

```diff
@@ -304,6 +310,48 @@ class="css_button link_submit" target='Main' onClick="top.restoreSession()">
 ?>
     </tr>
 
+ <tr>
+  <td nowrap>
+   <b><?php echo xlt('Provider'); ?>:</b>
+  </td>
+  <td nowrap>
+
+<?php
+ // Get the providers list which we will work off of.
+$ures = sqlStatement("SELECT id, username, fname, lname FROM users WHERE " .
+  "authorized != 0 AND active = 1 ORDER BY lname, fname");
+  if ($viewmode && ($result['provider_id'] > '0')) {
+        // Provider already given to us from a previously created encounter.
+        $defaultProvider = $result['provider_ID'];
+  } else {
+    // This is a new encounter
+    // default to the currently logged-in user
+    $defaultProvider = $_SESSION['authUserID'];
+    // get provider from existing event involving this patient on the session['lastcaldate'] used
+    $query = "SELECT pc_aid FROM openemr_postcalendar_events WHERE pc_eventDate = ? and pc_pid =?";
+    $qprov = sqlStatement ($query, array($_SESSION['lastcaldate'],$pid) );
+    if (count($qprov) > '0') { 
```

Comment:
Doesn't seem valid to use count() on a recordset. Perhaps you want to call sqlNumRows()?